pkg/store/mock/scanner.go

```diff
@@ -0,0 +1,191 @@
+package mock
+
+import (
+	"bytes"
+	"database/sql"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	ErrNilPtr      = errors.New("destination pointer is nil")
+	ErrUnknownType = errors.New("unknown type for conversion: reflection required")
+)
+
+// MockScanner allows testing `Scan` interfaces for models. You can add an error
+// to return or the data items you wish to scan into a model using `SetError()`
+// or `SetData()`.
+type MockScanner struct {
+	err        error
+	data       []any
+	scanned    int
+	notScanned int
+	testLog    []string
+}
+
+// Scan will return an error if set, otherwise it will check that the lengths of
+// `dest` and the `data` are equal and then attempt to scan each `data` item into
+// the `dest` item with the same index, returning any errors.
+func (m *MockScanner) Scan(dest ...any) (err error) {
+	// if the user set an error, return it
+	if m.err != nil {
+		return m.err
+	}
+
+	// ensure each data item can be scanned into the destination it's assigned to
+	for i, dst := range dest {
+		src := m.data[i]
+
+		// If the type is a scanner type then scan it directly.
+		if scanner, ok := dst.(sql.Scanner); ok {
+			if err = scanner.Scan(src); err != nil {
+				return fmt.Errorf("failed to scan data into destination at index %d: %w", i, err)
+			}
+			m.scanned++
+			continue
+		}
+
+		// Check the common cases without reflection.
+		if err = convertAssign(dst, src); err != nil {
+			if errors.Is(err, ErrUnknownType) {
+				m.notScanned++
+				m.logf("unsupported Scan, storing drive.Value type %T into type %T at index %d", src, dst, i)
```

Comment:
FYI the string format `%T` will print the type so that you don't have to use `reflect.TypeOf(dst).Kind().String()`